ndonnx/_core/_numericimpl.py

```diff
@@ -568,18 +568,17 @@ def cumulative_sum(
             else:
                 raise ValueError("axis must be specified for multi-dimensional arrays")
 
-        if dtype is None:
-            if isinstance(x.dtype, (dtypes.Unsigned, dtypes.NullableUnsigned)):
-                if ndx.iinfo(x.dtype).bits < 64:
-                    out = x.astype(dtypes.int64)
-                else:
-                    raise ndx.UnsupportedOperationError(
-                        f"Cannot perform `cumulative_sum` using {x.dtype}"
-                    )
+        if isinstance(x.dtype, (dtypes.Unsigned, dtypes.NullableUnsigned)):
+            if ndx.iinfo(x.dtype).bits < 64:
+                out = x.astype(dtypes.int64)
             else:
-                out = x.astype(_determine_reduce_op_dtype(x, dtype, dtypes.int64))
+                return NotImplemented
+        elif dtype == dtypes.uint64 or dtype == dtypes.nuint64:
+            raise ndx.UnsupportedOperationError(
+                f"Unsupported dtype parameter for cumulative_sum {dtype} due to missing kernel support"
+            )
         else:
-            out = out.astype(dtype)
+            out = x.astype(_determine_reduce_op_dtype(x, None, dtypes.int64))
```

Comment:
`maximum_unsigned_dtype=int64` (with a signed integer) seems hacky. Could be just the naming of the parameter

Reply:
I think it's just poor naming. I'll change it to `uint64` since it's all the same functionally.